docs/metrics.md

```diff
@@ -0,0 +1,161 @@
+  | thanos_cache_inmemory_max_size_bytes              | gauge     | Maximum number of bytes to be held in the inmemory cache.                                                       |
+  | thanos_cache_inmemory_total_size_bytes            | gauge     | Current byte size of items (both value and key) in the inmemory cache.                                          |
+  | thanos_cache_inmemory_max_item_size_bytes         | gauge     | Maximum number of bytes for single entry to be held in the inmemory cache.                                      |
+  | thanos_cache_inmemory_items_overflowed_total      | counter   | Total number of items that could not be added to the inmemory cache due to being too big.                       |
+  | thanos_cache_inmemory_requests_total              | counter   | Total number of requests to the inmemory cache.                                                                 |
+  | thanos_cache_memcached_hits_total                 | counter   | Total number of items requests to the cache that were a hit.                                                    |
+  | thanos_cache_memcached_requests_total             | counter   | Total number of items requests to memcached.                                                                    |
+  | thanos_cache_redis_hits_total                     | counter   | Total number of items requests to the cache that were a hit.                                                    |
+  | thanos_cache_redis_requests_total                 | counter   | Total number of items requests to redis.                                                                        |
+  | thanos_memcached_client_info                      | gauge     | A metric with a constant '1' value labeled by configuration options from which memcached client was configured. |
+  | thanos_memcached_operations_total                 | counter   | Total number of operations against memcached.                                                                   |
+  | thanos_memcached_operation_failures_total         | counter   | Total number of operations against memcached that failed.                                                       |
+  | thanos_memcached_operation_skipped_total          | counter   | Total number of operations against memcached that have been skipped.                                            |
+  | thanos_memcached_operation_duration_seconds       | histogram | Duration of operations against memcached.                                                                       |
+  | thanos_memcached_operation_data_size_bytes        | histogram | Tracks the size of the data stored in and fetched from memcached.                                               |
+  | thanos_redis_operation_duration_seconds           | histogram | Duration of operations against redis.                                                                           |
+
+### List of Metrics Exported By Thanos Sidecar Shipper
```

Comment:
If these are metrics of the Thanos Sidecar, they should be in the Sidecar's documentation.

Reply:
I will move the Shipper metrics to the Sidecar PR accordingly.